internal/pkg/docker/orchestrator/orchestrator.go

```diff
@@ -452,6 +485,80 @@ func (o *Orchestrator) waitForContainerToStart(ctx context.Context, id string) e
 	}
 }
 
+func (o *Orchestrator) waitForContainerDependencies(ctx context.Context, name string, a *runTaskAction) error {
+	var deps []string
+	for depName, state := range a.task.Containers[name].DependsOn {
+		deps = append(deps, fmt.Sprintf("%s->%s", depName, state))
+	}
+	logMsg := strings.Join(deps, ", ")
+	fmt.Printf("Waiting for container %q dependencies: [%s]\n", name, color.Emphasize(logMsg))
+	eg, ctx := errgroup.WithContext(ctx)
+	for name, state := range a.task.Containers[name].DependsOn {
+		name, state := name, state
+		eg.Go(func() error {
+			ctrId := o.containerID(name)
+			isEssential := a.task.Containers[name].IsEssential
+			ticker := time.NewTicker(700 * time.Millisecond)
+			defer ticker.Stop()
+			for {
+				select {
+				case <-ticker.C:
+				case <-ctx.Done():
+					return ctx.Err()
+				}
+				switch state {
+				case ctrStateStart:
+					err := o.waitForContainerToStart(ctx, ctrId, isEssential)
+					if err != nil {
+						return err
+					}
+					if isEssential {
+						log.Successf("Successfully started container %s\n", ctrId)
+						return nil
+					}
+					fmt.Printf("non-essential container %q started successfully\n", ctrId)
+					return nil
+				case ctrStateHealthy:
+					healthy, err := o.docker.IsContainerHealthy(ctx, ctrId)
+					if err != nil {
+						if !isEssential {
+							fmt.Printf("non-essential container %q is not healthy: %v\n", ctrId, err)
+							return nil
+						}
+						return fmt.Errorf("essential container %q is not healthy: %w", ctrId, err)
+					}
+					if healthy {
+						log.Successf("Successfully dependency container %q reached %q\n", ctrId, state)
+						return nil
+					}
+				case ctrStateComplete:
+					exitCode, err := o.docker.IsContainerCompleteOrSuccess(ctx, ctrId)
```

Comment:
same question! should we wait for it to complete?

Reply:
Same thing as `healthy` status
current implementation is waiting for container to exit by comparing the error returned by `containerExit code` 
https://github.com/aws/copilot-cli/pull/5509/files#diff-e21274f5c1fd77d8e5391a27014297ea2818cc1055df6c7a0eeff56e3251aebaR548-R551